iOS: support for iOS port

pinkLovepurple opened the pull request:
1, make it runnable on iOS
2, fix audio and replace some deprecated function like "AudioFileReadPackets"
3, add some bugs

now we can reach the title screen and start login
*** need lots of fixes on HTTP***

notice: you may replace my SIF stuff in Xcode with yours or it will fail on compiling

***UPDATE #1 ***
1, fix lua "imcompatible precompiled chunk" on 64bit-system
2, add support for iOS arm64
3, update libcurl and add relative full-platform dependencies
4, remove game-in purchase 
5, add crypto algorithm

pinkLovepurple committed:
fix decryption header size

now it can handle sif's file
but still not working out until fix opengl

pinkLovepurple committed:
Merge pull request #2 from kotori2/master

keep code update

pinkLovepurple committed:
Merge pull request #3 from kotori2/master

pull before update

pinkLovepurple committed:
make support for ios port

1, make it runnable on iOS
2, fix audio and replace some deprecated function like "AudioFileReadPackets"
3, add some bugs

now we can reach the title screen and start login
*** need lots of fixes on HTTP***

notice: you may replace my SIF stuff in Xcode with yours or it will fail on compiling

pinkLovepurple commented:
> Is that something missing and you removed lots of function calls

I need some time to add openssl for iOS

kotori2 commented:
Use platform-provided crypto libs if it is possible. OpenSSL is more than
4MB so it is definitely not a good choice.

miyamizu <notifications@github.com> 于 2020年1月15日周三 16:41写道：

> Is that something missing and you removed lots of function calls
>
> I need some time to add openssl for iOS
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/kotori2/Playground-SIF/pull/4?email_source=notifications&email_token=AEGYNSJXFJY2A7UV2TZQGCDQ53D4BA5CNFSM4KG5ZB2KYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEI7QFII#issuecomment-574554785>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AEGYNSOLGDJHQYJZNQO32FDQ53D4BANCNFSM4KG5ZB2A>
> .
>

pinkLovepurple committed:
Merge pull request #4 from kotori2/master

pull before update #2

pinkLovepurple committed:
solve imcompatible precompiled chunk on 64bit system

solve imcompatible precompiled chunk on 64bit system

pinkLovepurple committed:
tons of fixes

There are some fixes on iOS build
1: add support for iOS arm64
2: remove game-in purchase (you can deploy app on iOS device now)
3: add full-platform dependencies

pinkLovepurple committed:
Merge pull request #5 from kotori2/master

pull before update

pinkLovepurple committed:
iOS: add crypto algorithm

pinkLovepurple committed:
iOS: two fixes

1: completely remove "SKProducts" && "SKPayment"
2: temporarily fix crash on start login

kotori2 commented on Engine/libs/lua/loslib.c:
```diff
@@ -77,14 +77,15 @@
 
 
 static int os_execute (lua_State *L) {
-  const char *cmd = luaL_optstring(L, 1, NULL);
```
Add a #ifdef flag instead of remove it completely

pinkLovepurple replied:
got it

kotori2 commented on Engine/porting/iOS/CSockReadStream.cpp:
```diff
@@ -133,25 +133,25 @@ CSockReadStream::setStatus()
 bool
 CSockReadStream::readRingBuf()
 {
-    if((m_lastPos > m_getPos) && (m_lastPos < READ_BUFSIZ)) {
```
Why you commented all socket stuffs?

pinkLovepurple committed:
iOS: restore socket and misdeleting

pinkLovepurple committed:
Lua: add ifndef branch for Apple

hard to know whether it is mac or ios

pinkLovepurple committed:
iOS: fix libs path && add missing files

pinkLovepurple committed:
Engine: fix mp3 skip offset

pinkLovepurple committed:
Engine: fix mp3 header#2

it can play sound in mp3 formot now

Salaron commented on Engine/source/Sound/CSoundAnalysisMP3.cpp:
```diff
@@ -225,7 +225,7 @@ bool SoundAnalysis_MP3( const char* _path, sSoundAnalysisData* _analysisData )
         // フレームヘッダ読み込み
 		decryptor.gotoOffset(skipOffset);
-        fseek( fp, skipOffset + (decryptor.m_useNew ? 4 : 0), SEEK_SET );
+        fseek( fp, skipOffset + (decryptor.m_useNew ? 16 : 0), SEEK_SET );
```
it's better to use ```decryptor.m_header_size``` instead of ```(decryptor.m_useNew ? 16 : 0)```, because it will fail if file encrypted with 2 version

Salaron reviewed (approved):
Before merging it's better to add to .gitignore files of your IDE and remove it from git, which are in ```Engine/porting/iOS/Project/SampleProject/SampleProject.xcodeproj``` folder (if they are not needed for a successful build)

kotori2 commented:
> Before merging it's better to add to .gitignore files of your IDE and remove it from git, which are in `Engine/porting/iOS/Project/SampleProject/SampleProject.xcodeproj` folder (if they are not needed for a successful build)

I think xcodeproj folder is smh required by Xcode, gonna test it out later

Salaron commented:
> I think xcodeproj folder is smh required by Xcode, gonna test it out later

It contains ```xcuserdata``` folder, which have breakpoints list. If ```xcodeproj``` is required, then exclude only ```xcuserdata``` folder.

kotori2 commented:
@Salaron https://github.com/github/gitignore/blob/master/Global/Xcode.gitignore
Yes, seems just ignoring xcuserdata is enough.

Commit:
Update .gitignore

Commit:
iOS: add WebView headers

and some clean up

Commit:
Revert "Update .gitignore"

This reverts commit b2d28ce75f9c383807327a41e26fbfb18e018ded.

Commit:
Revert "Revert "Update .gitignore""

This reverts commit 8d34b17fb8cb591f64453767b54dc499fb18252f.

Commit:
clean up

kotori2 commented:
iOS compilation succeed on my device, but it seems encryption not working
```
request_data={"login_key": "AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA==","login_passwd": "AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA"}
```
Gonna merge anyway
